app/helpers/oauth.py

```diff
@@ -0,0 +1,21 @@
+from app.models.User import UserOAuthToken
+import requests
+from json import loads as json_parse
+from config import oauth
+
+# ?order=desc&sort=reputation&site=codegolf&filter=!T6o*9Tv*x)DWaQr(r5
+def stackexchange(auth_token):
+    user = requests.get(
+        'https://api.stackexchange.com/2.2/me',
+        params={
+            'site': 'codegolf',
+            'filter': '!)RwcIFN1JaCrhVpgyYeR_oO*', # Constant obtained from SE API explorer
+            'access_token': auth_token,
+            'key': oauth.get('se').get('key')
+        }
+    ).json().get('items')[0]
+    return user.get('user_id'), {
+        'name': user.get('display_name'),
+        'avatar': user.get('profile_image'),
+        'email':'NULL@example.com'
```

Comment:
Set email to `None` instead - `email` is allowed to be `NULL` in the db, and it's more meaningful than an email that looks valid

Reply:
I can do that but in the case the user unsets their avatar than a server error will occur loading their avatar as it will attempt to generate a gravatar but fail when email is `None`

Reply:
We can special-case the gravatar code to generate a random gravatar for `None` input